packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -191,37 +191,26 @@ export class RunBatchJob implements sfn.IStepFunctionsTask {
     }
 
     // validate arraySize limits
-    if (
-      props.arraySize !== undefined &&
-      (props.arraySize < 2 || props.arraySize > 10000)
-    ) {
-      throw new Error(
-        `Invalid value of arraySize. The array size can be between 2 and 10,000.`
-      );
+    if (props.arraySize !== undefined && !Token.isUnresolved(props.arraySize) &&
+      (props.arraySize < 2 || props.arraySize > 10_000)) {
```

Comment:
is `10_000` intended?

Reply:
Yes. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#numeric-separators

Reply:
I believe there is a validation routine in `core`, specifically for people writing TypeScript constructs, which will do `!isUnresolved` for you. Let me look it up.

Reply:
https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/core/lib/token.ts#L199

Reply:
> Yes. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#numeric-separators

neat